packages/salesforcedx-vscode-core/src/orgBrowser/metadataCmp.ts

```diff
@@ -84,79 +85,170 @@ export class ComponentUtils {
     }
   }
 
-  public async buildCustomObjectFieldsList(
-    objectName: string,
-    defaultUsernameOrAlias: string,
-    outputPath: string
-  ): Promise<string[]> {
-    const jsonResult = await forceListSchemaSobjectDescribe(
-      objectName,
-      defaultUsernameOrAlias,
-      outputPath
-    );
-
-    const result = JSON.parse(jsonResult);
-    const fields = result.result.fields.map(
-      (field: {
-        type: string;
-        relationshipName?: string;
-        name: string;
-        length?: number;
-      }) => {
-        switch (field.type) {
-          case 'string':
-          case 'textarea':
-          case 'email':
-            return `${field.name} (${field.type}(${field.length}))`;
-          case 'reference':
-            return `${field.relationshipName} (reference)`;
-          default:
-            return `${field.name} (${field.type})`;
-        }
+  public buildCustomObjectFieldsList(
+    sObject: string,
+    result?: string,
+    componentsPath?: string
+  ): string[] {
+    try {
+      if (isNullOrUndefined(result)) {
+        result = fs.readFileSync(componentsPath!, 'utf8');
       }
-    );
+      const jsonResult = JSON.parse(result);
+      const fields = jsonResult.result.map(
+        (field: {
+          name: string;
+          type: string;
+          relationshipName?: string;
+          length?: number;
+        }) => {
+          switch (field.type) {
+            case 'string':
+            case 'textarea':
+            case 'email':
+              return `${field.name} (${field.type}(${field.length}))`;
+            case 'reference':
+              return `${field.relationshipName} (reference)`;
+            default:
+              return `${field.name} (${field.type})`;
+          }
+        }
+      );
+      telemetryService.sendEventData('CustomObjects Fields quantity', { sObject }, { fields: fields.length });
```

Comment:
Great job adding telemetry. I think the field quantity won't be much help here. Can we instead add telemetry to see how many components we get in listMetadataTypes()? Also we want the event name to be on worded - something like orgBrowserMetadataFetched? This would show us this originated from orgBrowser?

Reply:
Removing the telemetry record as it's not adding that much of value to it.